apps/astarte_data_updater_plant/test/astarte_data_updater_plant/data_updater/core/device_test.exs

```diff
@@ -0,0 +1,130 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.DataUpdaterPlant.DataUpdater.Core.DeviceTest do
+  use Astarte.Cases.Data, async: true
+  use Mimic
+  import Astarte.Helpers.DataUpdater
+
+  alias Astarte.Core.Device
+  alias Astarte.DataUpdaterPlant.DataUpdater
+  alias Astarte.DataUpdaterPlant.DataUpdater.Core
+  alias Astarte.DataUpdaterPlant.RPC.VMQPlugin
+  alias Astarte.DataAccess.Devices.Device, as: DatabaseDevice
+  alias Astarte.DataAccess.Realms.Realm
+  alias Astarte.DataAccess.Repo
+  alias Astarte.Helpers.Database
+
+  @timestamp_us_x_10 Database.make_timestamp("2025-05-14T14:00:32+00:00")
+
+  setup do
+    device_id = Database.random_device_id()
+    encoded_device_id = Device.encode_device_id(device_id)
+    {:ok, device_id: device_id, encoded_device_id: encoded_device_id}
+  end
+
+  describe "ask_clean_session/2" do
+    test "disconnects device and clears cache on successful disconnect", %{
+      realm_name: realm_name,
+      device_id: device_id,
+      encoded_device_id: encoded_device_id
+    } do
+      # Insert initial device state
+      state = setup_device_state(realm_name, device_id, encoded_device_id, [])
```

Comment:
There exists the `Astarte.Cases.Device` ExUnit case for this! you can `use` it and it takes care of inserting a device with some interfaces for you, you can then use it to dump the state and have an actual state to work on
```elixir
  use Astarte.Cases.Device
  import Astarte.Helpers.DataUpdater

  setup_all %{realm_name: realm_name, device: device} do
    setup_data_updater(realm_name, device.encoded_id)
    state = DataUpdater.dump_state(realm_name, device.encoded_id)

    %{state: state}
  end
```